File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        int max1 = Integer.MAX_VALUE;  // The smallest number in the triplet
        int max2 = Integer.MAX_VALUE;  // The second smallest number in the triplet

        // Iterate through the array
        for (int n : nums) {
            if (n <= max1) {
                // n is smaller than or equal to max1, update max1
                max1 = n;
            } else if (n <= max2) {
                // n is greater than max1 but smaller than or equal to max2, update max2
                max2 = n;
            } else {
                // n is greater than both max1 and max2, a triplet is found
                return true;  // Early exit
            }
        }

        // If no increasing triplet is found
        return false;
    }
}

File: 0334-increasing-triplet-subsequence/0334-increasing-triplet-subsequence.py
class Solution:
    def increasingTriplet(self, nums: List[int]) -> bool:
        # Initialize two variables to track the smallest (mi) and the second smallest (mid) values
        mi, mid = float('inf'), float('inf')
        
        # Iterate through each number in the input list
        for num in nums:
            # If the current number is greater than the second smallest, we found an increasing triplet
            if num > mid:
                return True
            # If the current number is less than or equal to the smallest, update the smallest
            if num <= mi:
                mi = num
            # If the current number is between the smallest and the second smallest, update the second smallest
            else:
                mid = num
        
        # If no increasing triplet is found, return False
        return False
